sdgym/datasets.py

```diff
@@ -62,94 +95,161 @@ def _get_dataset_path(
     if dataset.exists():
         return dataset
 
-    datasets_path = datasets_path or DATASETS_PATH
+    datasets_path = datasets_path or DATASETS_PATH / modality
     dataset_path = datasets_path / dataset
-    if dataset_path.exists():
+    if dataset_path.exists() and _path_contains_data_and_metadata(dataset_path):
         return dataset_path
 
     bucket = bucket or BUCKET
     if not bucket.startswith(S3_PREFIX):
-        local_path = Path(bucket) / dataset if bucket else Path(dataset)
-        if local_path.exists():
+        local_path = Path(bucket) / modality / dataset if bucket else Path(dataset)
+        if local_path.exists() and _path_contains_data_and_metadata(local_path):
             return local_path
```

Comment:
why does this case even exist?

Reply:
This was some legacy code that we had where we consider a 'bucket' something like an offline folder (offline bucket)... I can remove it, but wanted to keep it as 'close' as possible to the previous functionality.